db/data/20240423162700_create_server_tags_option.rb

```diff
@@ -0,0 +1,61 @@
+# BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
+#
+# Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
+#
+# This program is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License as published by the Free Software
+# Foundation; either version 3.0 of the License, or (at your option) any later
+# version.
+#
+# Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Greenlight; if not, see <http://www.gnu.org/licenses/>.
+
+# frozen_string_literal: true
+
+class CreateServerTagsOption < ActiveRecord::Migration[7.0]
+  def up
+    MeetingOption.create!(name: 'serverTag', default_value: '') unless MeetingOption.exists?(name: 'serverTag')
+    tag_option = MeetingOption.find_by!(name: 'serverTag')
+    unless RoomsConfiguration.exists?(meeting_option: tag_option, provider: 'greenlight')
+      RoomsConfiguration.create!(meeting_option: tag_option, value: 'optional', provider: 'greenlight')
+    end
+    Tenant.all.each do |tenant|
+      unless RoomsConfiguration.exists?(meeting_option: tag_option, provider: tenant.name)
+        RoomsConfiguration.create!(meeting_option: tag_option, value: 'optional', provider: tenant.name)
+      end
+    end
+    Room.find_each { |room| RoomMeetingOption.find_or_create_by!(room:, meeting_option: tag_option) }
+
+    MeetingOption.create!(name: 'serverTagRequired', default_value: 'false') unless MeetingOption.exists?(name: 'serverTagRequired')
+    tag_required_option = MeetingOption.find_by!(name: 'serverTagRequired')
+    unless RoomsConfiguration.exists?(meeting_option: tag_required_option, provider: 'greenlight')
+      RoomsConfiguration.create!(meeting_option: tag_required_option, value: 'optional', provider: 'greenlight')
+    end
+    Tenant.all.each do |tenant|
+      unless RoomsConfiguration.exists?(meeting_option: tag_required_option, provider: tenant.name)
+        RoomsConfiguration.create!(meeting_option: tag_required_option, value: 'optional', provider: tenant.name)
+      end
+    end
+    Room.find_each do |room|
+      unless RoomMeetingOption.exists?(room:, meeting_option: tag_required_option)
+        RoomMeetingOption.create!(room:, meeting_option: tag_required_option, value: 'false')
+      end
+    end
```

Comment:
This definitely needs to be cleaned up and optimized for better performance. `find_each` for large deployments will cause the deployment to crash. We're also looping twice over all Rooms when we can loop once. Here's some suggested improvements (in kinda pseudo code)

```
create serverTag and serverTagRequired meeting options
create serverTag and serverTagRequired RoomConfigurations
create serverTag and serverTagRequired RoomConfigurations for all tenants

Room.find_in_batches |group|
  group.each do |room|
    create serverTag and serverTagRequired RoomMeetingOption
end
```

Reply:
@farhatahmad I'm working on it right now and am testing it on our largest database.
Already one thing FYI: find_each does in fact use a batch size of 1000 per default. It's not like find_all.

Reply:
With the latest commit [ca4217b](https://github.com/bigbluebutton/greenlight/pull/5790/commits/ca4217b0e37a4d9462927cad18361f516e171767) the migration will only take about a minute where it took over 2 hours before and still retains the "safe but slow"  variant if existing tag options are detected.